### PR TITLE
test: raise active-runner.ts and mcp-server.ts coverage to 85%+ (WOP-1930)

### PR DESCRIPTION
**Pull request by AgentWOPR**
## Summary
Closes WOP-1930

- Added 14 new tests to `tests/execution/active-runner.test.ts` covering `parseResponse` null/missing branches, `resolveModel` fallbacks, `run()` flowName resolution, `pollOnce` all-fail, `processInvocation` error paths, abort signal handling, and `sleep()` pre-aborted exit
- Added 37 new tests to `tests/execution/mcp-server.test.ts` covering admin CRUD handlers (flow/state/transition/gate create+update), `admin.flow.snapshot`, `admin.flow.restore`, `flow.report` engine-unavailable and processSignal-throws paths, `flow.claim` race conditions, and affinity set/failure paths

## Test plan
- [x] `npm run check` passes (biome + tsc clean)
- [x] `npx vitest run tests/execution/active-runner.test.ts` — 25 tests pass
- [x] `npx vitest run tests/execution/mcp-server.test.ts` — 98 tests pass

Generated with Claude Code

---

<a href="https://app.devin.ai/review/wopr-network/defcon/pull/94" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

## Summary by Sourcery

Expand test coverage for active-runner and MCP server execution paths to improve reliability and error handling guarantees.

Tests:
- Add extensive tests for ActiveRunner covering response parsing, model resolution fallbacks, polling behavior, invocation error paths, and abort/sleep signal handling.
- Add comprehensive MCP server admin and flow tool handler tests, including CRUD flows/states/transitions/gates, snapshot/restore, flow reporting, claiming behavior, and affinity handling edge cases.

> [!NOTE]
> ### Raise test coverage to 85%+ and add failure-handling in `execution.ActiveRunner.processInvocation` in [active-runner.ts](https://github.com/wopr-network/defcon/pull/94/files#diff-6043fdb04254c13280edc57e4026c1ef02f51b6d92bc62a75a34ef455a17fe7d)
> Add a nested try/catch in `execution.ActiveRunner.processInvocation` to call `invocationRepo.releaseClaim(invocation.id)` when `complete()` throws. Expand tests for runner flow, claim failures, abort handling, and admin tool handlers for MCP server.
>
> #### 📍Where to Start
> Start with `processInvocation` in [active-runner.ts](https://github.com/wopr-network/defcon/pull/94/files#diff-6043fdb04254c13280edc57e4026c1ef02f51b6d92bc62a75a34ef455a17fe7d), then review new tests in [tests/execution/active-runner.test.ts](https://github.com/wopr-network/defcon/pull/94/files#diff-8e96e19a8f320dd16c39163d4c4a1fa78675adc0c5465f297742976febcc92af) and [tests/execution/mcp-server.test.ts](https://github.com/wopr-network/defcon/pull/94/files#diff-8fead31125804af6284938380525bc4d7da0a953705fd1fb322c3b3db076c603).
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 73ceaff.</sup>
> 
>
>

### CODE REVIEW

**Comment by AgentWOPR** [comment]
⚠️ **wopr-auto review: ISSUES FOUND.**

Qodo has an open /improve suggestion (importance 6, Low) at `src/execution/active-runner.ts [115-117]` suggesting to wrap `releaseClaim()` in a try/catch block. The current diff already implements this pattern, suggesting the Qodo suggestion may be stale or based on an intermediate commit state. Devin has confirmed the fix is already in place.

Per review policy, open Qodo /improve suggestions are treated as blocking. The fixer should either resolve the Qodo suggestion inline or confirm it is stale before merge.

**Comment by AgentWOPR** [comment]
⚠️ **wopr-auto review: ISSUES FOUND.**

Qodo has 3 open /improve suggestions (persistent comment #4017970917, last updated to 66730d3) that remain unaddressed:

1. **[src/execution/active-runner.ts:114-125] Medium (importance 7) — Preserve error stack traces**
   Log `releaseErr` directly instead of `releaseErr.message` to preserve stack traces.

2. **[tests/execution/mcp-server.test.ts:1297-1302] Low (importance 5) — Robust success assertion**
   Use `expect(result.isError).not.toBe(true)` instead of `.toBeUndefined()`.

3. **[tests/execution/active-runner.test.ts:332-366] Low (importance 6) — Private method access**
   Tests call `runner.parseResponse()` directly but `parseResponse` is private — this will cause TypeScript compilation errors. Use `(runner as any).parseResponse()`.

Fixer should address these before this PR can be approved.

File: tests/execution/active-runner.test.ts [hunk 1]
```diff
@@ -329,6 +329,281 @@ describe("maxRetries guard on processSignal error", () => {
...
+
+    const runner = new ActiveRunner(deps);
+    await runner.run({ once: true });
+
+    expect(processSignalMock).not.toHaveBeenCalled();
+    expect(errorSpy).toHaveBeenCalledWith(
+      expect.stringContaining("complete() failed"),
+      expect.any(Error),
+    );
+    errorSpy.mockRestore();
```
[the remark this answers is not in the document]
Comment (AgentWOPR) [reply tests/execution/active-runner.test.ts]: This is already resolved — the catch block for `complete()` failure now calls `releaseClaim()` in its own nested try/catch (logging but not rethrowing if that also fails). The test at line 565 asserts `releaseClaimMock.toHaveBeenCalledWith(invocation.id)`. Suggestion is based on an earlier commit before fixer-1930-3 landed the fix.
